source/base/hdf5.cc

```diff
@@ -105,18 +106,74 @@ namespace HDF5
     , global_no_collective_cause(H5D_MPIO_SET_INDEPENDENT)
   {
     hdf5_reference = std::shared_ptr<hid_t>(new hid_t, [](hid_t *pointer) {
-      // Release the HDF5 resource
-      const herr_t ret = H5Dclose(*pointer);
-      AssertNothrow(ret >= 0, ExcInternalError());
-      (void)ret;
-      delete pointer;
+#  ifdef DEAL_II_WITH_MPI
+#    if __cpp_lib_uncaught_exceptions >= 201411
```

Comment:
We now require C++14, so you can remove this `#ifdef` here and the whole `#else` part. That will make things simpler to read too.